mne/encoding/feature.py

```diff
@@ -0,0 +1,250 @@
+"""A collection of classes for transforming, preprocessing, and fitting."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy import sparse
+from ..utils import warn
+
+
+class FeatureDelayer(object):
+    """Transformer for creating delayed versions of an input stimulus.
+
+    Parameters
+    ----------
+    time_window : array, shape (2,)
+        Specifies a minimum/maximum delay in seconds. In this case, all
+        delays between `min(time_window)` and `max(time_window)`
+        will be calculated using `sfreq`.
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future.
```

Comment:
See the latest version of the code, which includes support for both cases in a single `delay_timeseries` function. In this case, you can supply `delays=` and give an explicit set of floats, or `window=` and give a tmin/tmax tuple/list. In addition, both cases allow for you to give a list of delays, or a list of windows, and when the `transform` method is called, it will use a different set of delays / windows for each feature. Would that work?